provider/source/src/cldr_serde/units/info.rs

```diff
@@ -80,3 +81,29 @@ pub(crate) struct Supplemental {
 pub(crate) struct Resource {
     pub(crate) supplemental: Supplemental,
 }
+
+impl Resource {
+    pub fn get_unit_id(&self, unit_name: &str) -> Result<u16, DataError> {
+        self.supplemental
+            .convert_units
+            .convert_units
+            .iter()
+            .enumerate()
+            .find(|(_, (name, _))| name.as_str() == unit_name)
+            .map(|(index, _)| index)
+            .ok_or_else(|| DataError::custom("Unit not found"))
+            .and_then(|index| {
+                u16::try_from(index).map_err(|_| DataError::custom("Index out of range for u16"))
+            })
+    }
+
+    pub fn get_unit_ids_map(&self) -> BTreeMap<String, usize> {
```

Comment:
this should return `u16`s as well

Reply:
and also be used in the other places where you use ids

Reply:
I prefer to return `usize` because we do not return an error with it, if you insist, then we need to return `Result`

Reply:
yes it should return a `Result`

Reply:
done

Reply:
skip this for now as discussed online:  https://github.com/unicode-org/icu4x/pull/6620